new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -494,6 +738,189 @@ void tearDown() {
     assertDoesNotThrow(() -> deleteDomainCustomResource(domainUid, domainNamespace1),
             "deleteDomainCustomResource failed with ApiException");
     logger.info("Deleted Domain Custom Resource " + domainUid + " from " + domainNamespace1);
+
+    // delete the domain images created in the test class
+    if (miiImagePatchAppV2 != null) {
+      deleteImage(miiImagePatchAppV2);
+    }
+    if (miiImageAddSecondApp != null) {
+      deleteImage(miiImageAddSecondApp);
+    }
+  }
+
+  private void pushImageIfNeeded(String image) {
+    // push the image to a registry to make the test work in multi node cluster
+    if (!REPO_USERNAME.equals(REPO_DUMMY_VALUE)) {
```

Comment:
This is broken too.

Reply:
You should login if REPO_USERNAME is set (to a non-dummy value), but you should push if REPO_NAME is non-empty. (Look at the code in other tests)

Reply:
fixed.